py3status/modules/getjson.py

```diff
@@ -68,28 +65,17 @@ class Py3status:
     def getjson(self):
         """
         """
-        response = {
-            'cached_until': self.py3.time_in(self.cache_timeout),
-        }
-
         try:
-            resp = self.py3.request(self.url, timeout=self.timeout)
-            status = resp.status_code == 200
-            resp = resp.json()
+            getjson = self.py3.request(self.url, timeout=self.timeout).json()
+            getjson = self.py3.flatten_dict(
+                getjson, delimiter=self.delimiter, intermediates=True)
         except self.py3.RequestException:
-            resp = None
-            status = False
-
-        if status:
-            response['full_text'] = self.py3.safe_format(
-                self.format,
-                self.py3.flatten_dict(
-                    resp, delimiter=self.delimiter, intermediates=True
-                )
-            )
-        else:
-            response['full_text'] = ''
-        return response
+            getjson = None
```

Comment:
From what I understand... I'm returning getjson value right away without any checking.

```
# Direct return
return {
    'cached_until': self.py3.time_in(self.cache_timeout),
    'full_text': self.py3.safe_format(self.format, getjson)
}
```
... TO...
```
# Quick check before returning
if getjson:
    getjson = self.py3.safe_format(self.format, getjson)
else:
    getjson = self.py3.safe_format(self.format, None)

return {
    'cached_until': self.py3.time_in(self.cache_timeout),
    'full_text': getjson
}
```
We have to check this for all modules that could come back None.
And maybe we have to check for (some) placeholders that could come back None.

This feels like we're making things too noice for unknown placeholders.

Need to play with this to see which feels right with `eth0` up/down.
For other known placeholders, it's not an issue... and it also tells us that some placeholders may have no values at some times. I think `[{}]`. Idk.
```
getjson{
    url = 'http://ip-api.com/json'
    format = 'GETJSON: [{lat}, ]{lon} {wrong}'
}
```